File: 27.remove-element.java
/*
 * @lc app=leetcode id=27 lang=java
 *
 * [27] Remove Element
 */

// @lc code=start
class Solution {
    public int removeElement(int[] nums, int val) {
        if (nums.length == 0) {
            return 0;
        }
        if (nums.length == 1) {
            return (nums[0] == val) ? 0 : 1;
        }
        int fast = 0, slow = 0;
        while (fast != nums.length) {
            if (nums[fast] != val) {
                nums[slow++] = nums[fast];
            }
            fast += 1;
        }
        return slow;
    }
}
// @lc code=end

File: 27.remove-element.cpp
/*
 * @lc app=leetcode id=27 lang=cpp
 *
 * [27] Remove Element
 */

// @lc code=start

#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        const auto pos = copy_if(nums.begin(), nums.end(), nums.begin(), [&val](const auto &el) {
            return el != val;
        });
        nums.erase(pos, nums.end());
        return nums.size();
    }
};
// @lc code=end
